packages/uploadthing/src/effect-http.ts

```diff
@@ -0,0 +1,114 @@
+import * as Http from "@effect/platform/HttpServer";
+import * as Effect from "effect/Effect";
+
+import type { Json } from "@uploadthing/shared";
+import {
+  FetchContext,
+  getStatusCodeFromError,
+  UploadThingError,
+} from "@uploadthing/shared";
+
+import { UPLOADTHING_VERSION } from "./internal/constants";
+import { formatError } from "./internal/error-formatter";
+import {
+  buildPermissionsInfoHandler,
+  buildRequestHandler,
+} from "./internal/handler";
+import { incompatibleNodeGuard } from "./internal/incompat-node-guard";
+import { toWebRequest } from "./internal/to-web-request";
+import type { FileRouter, RouteHandlerOptions } from "./internal/types";
+import type { CreateBuilderOptions } from "./internal/upload-builder";
+import { createBuilder } from "./internal/upload-builder";
+
+export { UTFiles } from "./internal/types";
+export type { FileRouter };
+
+type MiddlewareArgs = {
+  req: Http.request.ServerRequest;
+  res: undefined;
+  event: undefined;
+};
+
+export const createUploadthing = <TErrorShape extends Json>(
+  opts?: CreateBuilderOptions<TErrorShape>,
+) => createBuilder<MiddlewareArgs, TErrorShape>(opts);
+
+export const createRouteHandler = <TRouter extends FileRouter>(
+  opts: RouteHandlerOptions<TRouter>,
+): Http.router.Router<Http.body.BodyError, never> => {
+  incompatibleNodeGuard();
+
+  const requestHandler = buildRequestHandler<TRouter, MiddlewareArgs>(
+    opts,
+    "effect-http",
```

Comment:
note that `effect-http` [is a real thing](https://github.com/sukovanej/effect-http), we refer to the more low-level http server utilities as `@effect/platform`